Yosemite/Yosemite/Stores/CardPresentPaymentStore.swift

```diff
@@ -73,9 +97,28 @@ public final class CardPresentPaymentStore: Store {
 // MARK: - Services
 //
 private extension CardPresentPaymentStore {
+    /// Which backend is the store to use? WCPay or Stripe?
+    ///
+    enum CardPresentPaymentStoreBackend {
+        /// Use WCPay as the backend
+        ///
+        case wcpay
+
+        /// Use Stripe as the backend
+        ///
+        case stripe
+    }
+
     func startCardReaderDiscovery(siteID: Int64, onReaderDiscovered: @escaping (_ readers: [CardReader]) -> Void, onError: @escaping (Error) -> Void) {
         do {
-            try cardReaderService.start(WCPayTokenProvider(siteID: siteID, remote: self.remote))
+            switch usingBackend {
```

Comment:
You mentioned in the description making both remotes implement a common interface in a future PR.

I would take that one step further, and I would suggest extracting the logic to decide which backend we are calling into to a separate entity. For example, that could be a class with a single public method similar to this (naming is terrible, but just to illustrate the idea):
```
func getRemote(gateway: PaymentGateway) ->  PaymentsRemote {
switch gateway {
case .stripe: return stripeRemote
case .wcpay: return wcpayRemote
}
}
```

This class could be declared within this store, so it is completely private.

Reply:
Opened https://github.com/woocommerce/woocommerce-ios/issues/5904